### PR TITLE
Support using overwrite args

### LINKED ISSUE #7006: Control UniPS on TiFlash's side

### PR DESCRIPTION
**Pull request by CalvinNeo**
### What problem does this PR solve?

Issue Number: close #7006

Problem Summary:

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No code
image

image

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: dbms/src/Server/Server.cpp [hunk 1]
```diff
@@ -860,12 +868,39 @@ int Server::main(const std::vector<std::string> & /*args*/)
...
+    TiFlashStorageConfig storage_config;
+    // Deprecated settings.
+    // `global_capacity_quota` will be ignored if `storage_config.main_capacity_quota` is not empty.
+    // "0" by default, means no quota, the actual disk capacity is used.
+    size_t global_capacity_quota = 0;
+    std::tie(global_capacity_quota, storage_config) = TiFlashStorageConfig::parseSettings(config(), log);
+    if (storage_config.format_version)
+    {
+        setStorageFormat(storage_config.format_version);
+        LOG_INFO(log, "Using format_version={} (explicit stable storage format detected).", storage_config.format_version);
```
Comment (JaySon-Huang) [inline dbms/src/Server/Server.cpp:881]:
```suggestion
        LOG_INFO(log, "Using format_version={} (explicit storage format detected).", storage_config.format_version);
```

File: dbms/src/Server/Server.cpp [hunk 2]
```diff
@@ -860,12 +868,39 @@ int Server::main(const std::vector<std::string> & /*args*/)
...
 
+    // Init Proxy's config
+    TiFlashProxyConfig proxy_conf(config());
     EngineStoreServerWrap tiflash_instance_wrap{};
     auto helper = GetEngineStoreServerHelper(
         &tiflash_instance_wrap);
 
+    if (STORAGE_FORMAT_CURRENT.page == PageFormat::V4)
+    {
+        LOG_INFO(log, "use unips for proxy");
```
Comment (JaySon-Huang) [inline dbms/src/Server/Server.cpp:896]:
```suggestion
        LOG_INFO(log, "Using unips for proxy");
```

File: dbms/src/Server/Server.cpp [hunk 3]
```diff
@@ -860,12 +868,39 @@ int Server::main(const std::vector<std::string> & /*args*/)
...
+        proxy_conf.addExtraArgs("unips-enabled", "1");
+    }
+    else
+    {
+        LOG_INFO(log, "not use unips for proxy current {}", STORAGE_FORMAT_CURRENT.page);
```
Comment (JaySon-Huang) [inline dbms/src/Server/Server.cpp:901]:
```suggestion
        LOG_INFO(log, "unips is not enabled for proxy, page_version={}", STORAGE_FORMAT_CURRENT.page);
```

**Review by JaySon-Huang — APPROVED** [review]
Rest LGTM

**Review by lidezhu — APPROVED** [review]

**Pull request merged**